bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/discovery/ElroConnectsDiscoveryService.java

```diff
@@ -106,14 +108,15 @@ protected void startBackgroundDiscovery() {
     protected void stopBackgroundDiscovery() {
         logger.debug("Stop device background discovery");
         ScheduledFuture<?> job = discoveryJob;
-        if (job != null) {
+        if (job != null && !job.isCancelled()) {
             job.cancel(true);
             discoveryJob = null;
         }
     }
 
     @Override
     public void deactivate() {
+        removeOlderResults(new Date().getTime());
```

Comment:
Newer API can be used:
```suggestion
        removeOlderResults(Instant.now().toEpochMilli());
```

Reply:
Also adjusted in ElroConnectsBridgeDiscoveryService